src/js/components/FrameworkConfigurationForm.js

```diff
@@ -0,0 +1,329 @@
+      Object.keys(formData).forEach(property => {
+        this.writeErrors(
+          formData[property],
+          schema.properties[property],
+          schema.required && schema.required.includes(property),
+          errors[property]
+        );
+      });
+
+      return false;
+    }
+
+    if (isRequired && formData === "") {
+      errors.addError(`Expecting a ${schema.type} here, eg: ${schema.default}`);
+    }
+  }
+
+  getTotalErrorsForLevel(errorSchemaLevel) {
+    if (!errorSchemaLevel.__errors) {
+      return 0;
+    }
+    // only show one error per field in badge count
+    const currentLevelErrors = errorSchemaLevel.__errors.length > 0 ? 1 : 0;
+
+    return Object.keys(errorSchemaLevel)
+      .map(key => this.getTotalErrorsForLevel(errorSchemaLevel[key]))
+      .reduce((a, b) => a + b, currentLevelErrors);
+  }
+
+  handleFormChange(form) {
+    const { formData, errorSchema } = form;
+
+    const formErrors = {};
+    Object.keys(errorSchema).forEach(tab => {
+      formErrors[tab] = this.getTotalErrorsForLevel(errorSchema[tab]);
+    });
+
+    this.props.onFormErrorChange(formErrors);
+    this.props.onFormDataChange(formData);
+    this.setState({ errorSchema });
+  }
+
+  validate(formData, errors) {
+    const { packageDetails } = this.props;
+
+    this.writeErrors(formData, packageDetails.getConfig(), false, errors);
+
+    return errors;
+  }
+
+  formatErrorForJSON(errorSchema, path) {
+    if (!errorSchema.__errors) {
+      return [];
+    }
+    const currentLevelErrors = errorSchema.__errors.map(message => {
+      return { message, path };
+    });
+
+    return Object.keys(errorSchema)
+      .map(key => this.formatErrorForJSON(errorSchema[key], path.concat(key)))
+      .reduce((a, b) => a.concat(b), currentLevelErrors);
+  }
+
+  getErrorsForJSONEditor() {
+    const { errorSchema } = this.state;
+
+    if (!errorSchema) {
+      return [];
+    }
+
+    return this.formatErrorForJSON(errorSchema, []);
+  }
+
+  // transforms the errors from validating against the JSON schema
+  transformErrors(errorSchema) {
+    const errorSchemaCopy = [];
+
+    errorSchema.map(error => {
+      // remove name: "type" errors because caught by the validate() function
+      if (error.name === "type") {
+        return;
+      }
+
+      // capitalize other errors (ex: regex errors
+      error.message = StringUtil.capitalize(error.message);
+      errorSchemaCopy.push(error);
+    });
+
+    return errorSchemaCopy;
+  }
+
+  render() {
+    const {
+      packageDetails,
+      jsonEditorActive,
+      formData,
+      activeTab,
+      deployErrors
+    } = this.props;
+
+    const TitleField = props => {
+      return (
+        <h2 className="flush-top short-bottom">
+          {this.getFormattedSectionLabel(props.title)}
+        </h2>
+      );
+    };
+
+    const jsonEditorPlaceholderClasses = classNames(
+      "modal-full-screen-side-panel-placeholder",
+      { "is-visible": jsonEditorActive }
```

Comment:
There seems to be an issue with the JsonEditor toggle 

![json-editor-toggle](https://user-images.githubusercontent.com/180432/31638110-afa0f6fa-b286-11e7-9165-c0066a10b94a.gif)

Reply:
I can't reproduce this, but I found another issue with the JSON editor and tabs on Firefox. I will test on IE / Edge this morning. This form/tabs/json-editor is equivalent to the marathon app form, with the only different being the form tabs are `position: fixed` in this framework form.

Reply:
This is fixed now, with the cross-browser updates!